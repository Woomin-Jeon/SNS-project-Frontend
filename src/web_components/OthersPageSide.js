import React from 'react';

function OthersPageSide({
  loginState,
  specificPost,
}) {
  const { users } = loginState;
  const index = users.findIndex((user) => user.id === specificPost.id);

  return (
    <div className="mypage-side-banner">
      <div>
        <i className="fas fa-globe-americas" />
        <span className="mypage-side-banner-introduce">소개</span>
        <div className="mypage-side-banner-utter">
          _______________________________________________
        </div>
        <div className="mypage-user-information">
          <i className="fas fa-birthday-cake" />
          <div>{users[index].birth}</div>
          {' '}
          <br />
          <i className="fas fa-map-marker-alt" />
          <div>{users[index].location}</div>
          {' '}
          <br />
          <i className="far fa-envelope" />
          <div>{users[index].email}</div>
        </div>
      </div>
    </div>
  );
}

export default OthersPageSide;
