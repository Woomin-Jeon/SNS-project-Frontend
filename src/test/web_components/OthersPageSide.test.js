import OthersPageSide from "../../web_components/OthersPageSide";
import React from "react";
import {shallow} from 'enzyme';

describe('OthersPageSide', () => {
  let loginState;
  let specificPost;

  beforeEach(() => {
    loginState = {
      users: [
        {
          id: 'Random_ID',
          birth: '1996-04-21',
          location: '대한민국',
          email: 'dal96k@hanmail.net'
        }
      ]
    };
    specificPost = {
      id: 'Random_ID'
    }
  });

  it('renders well', () => {
    const component = shallow(
      <OthersPageSide
        loginState={loginState}
        specificPost={specificPost}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
