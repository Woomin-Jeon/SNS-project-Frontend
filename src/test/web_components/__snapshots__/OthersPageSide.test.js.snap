// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OthersPageSide renders well 1`] = `ShallowWrapper {}`;
